crates/network/src/native/client.rs

```diff
@@ -119,33 +110,32 @@ impl ElementComponent for GameClientView {
                 control_tx.send(Control::Disconnect).ok();
             }
         });
+        let gpu2 = Arc::clone(&gpu);
 
         // Run game logic
         {
-            let game_state = game_state.clone();
             let render_target = render_target.clone();
             let world_event_reader = Mutex::new(hooks.world.resource(world_events()).reader());
 
-            let game_client_exists = game_client.is_some();
+            let client_state = client_state.clone();
             hooks.use_frame(move |app_world| {
-                if !game_client_exists {
-                    return;
-                }
+                if let Some(client_state) = &client_state {
+                    let mut game_state = client_state.game_state.lock();
```

Comment:
Will anything be trying to lock this at the same time?

Reply:
Not sure. The `client_state` is shared quite a lot and a bit all over the place.

Haven't looked into how the wasm scripting uses it, but I suspect it may lock it as well

Reply:
Will look at it 👍